addons/website_mass_mailing/tests/test_mailing_website.py

```diff
@@ -0,0 +1,76 @@
+from odoo.addons.mail.tests.common import MockEmail
+from odoo.addons.mass_mailing.tests.common import MassMailCommon
+from odoo.tests.common import users
+
+import re
+
+
+class TestMassMailingWebsite(MassMailCommon, MockEmail):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestMassMailingWebsite, cls).setUpClass()
+        website1 = cls.env['website'].create({
+            'name': 'Website 1',
+            'domain': 'www.website1.com',
+        })
+        website2 = cls.env['website'].create({
+            'name': 'Website 2',
+            'domain': 'www.website2.com',
+        })
+        body_html = """
+            Test body
+            <a role="button" href="/unsubscribe_from_list" class="btn btn-link ">Unsubscribe</a>
+            <a href="/view" class="o_default_snippet_text">View Online</a>
+            <a role="button" href="/contactus" class="btn btn-link">Contact</a>
+        """
+        mailing_list = cls._create_mailing_list_of_x_contacts(1)
+        cls.mailing1 = cls.env['mailing.mailing'].create({
+            'name': 'Test Mailing 1',
+            'subject': 'Test Subject 1',
+            'body_html': body_html,
+            'mailing_model_id': cls.env.ref('mass_mailing.model_mailing_list').id,
+            'mailing_type': 'mail',
+            'contact_list_ids': [mailing_list.id],
+            'website_id': website1.id,
+        })
+        cls.mailing2 = cls.env['mailing.mailing'].create({
+            'name': 'Test Mailing 2',
+            'subject': 'Test Subject 2',
+            'body_html': body_html,
+            'mailing_model_id': cls.env.ref('mass_mailing.model_mailing_list').id,
+            'mailing_type': 'mail',
+            'contact_list_ids': [mailing_list.id],
+            'website_id': website2.id,
+        })
+
+    @users('user_marketing')
+    def test_mass_mailing_website(self):
```

Comment:
Maybe you can add an assertion about the link tracker created (just by searching the link tracker linked to the mailing and that the target URL is the "contactus")